.github/workflows/codeql.yml

```diff
@@ -0,0 +1,33 @@
+name: 'CodeQL'
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  schedule:
```

Comment:
Do we need the `schedule` and `workflow_dispatch`?

What is the use cases for them?

Reply:
Well, the schedule is what it literally means: run the workflow on a schedule. You can adapt this or remove it.

`workflow_dispatch` allows to run the workflow on demand, it was recently added on Actions as a feature.

Reply:
Sorry, let me clarify - do they provide added value assuming we run this for each PR and push to `main`?